src/stream/src/executor/temporal_join.rs

```diff
@@ -400,57 +412,117 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> TemporalJoinExecutor
                     yield Message::Watermark(watermark.with_idx(output_watermark_col_idx));
                 }
                 InternalMessage::Chunk(chunk) => {
-                    let mut builder = JoinStreamChunkBuilder::new(
-                        self.chunk_size,
-                        self.info.schema.data_types(),
-                        left_map.clone(),
-                        right_map.clone(),
-                    );
-                    let epoch = prev_epoch.expect("Chunk data should come after some barrier.");
-                    let keys = K::build(&self.left_join_keys, chunk.data_chunk())?;
-                    let to_fetch_keys = chunk
-                        .visibility()
-                        .iter()
-                        .zip_eq_debug(keys.iter())
-                        .filter_map(|(vis, key)| if vis { Some(key) } else { None });
-                    self.right_table
-                        .fetch_or_promote_keys(to_fetch_keys, epoch)
-                        .await?;
-                    for (r, key) in chunk.rows_with_holes().zip_eq_debug(keys.into_iter()) {
-                        let Some((op, left_row)) = r else {
-                            continue;
-                        };
-                        if key.null_bitmap().is_subset(&null_matched)
-                            && let join_entry = self.right_table.force_peek(&key)
-                            && !join_entry.is_empty()
-                        {
-                            for right_row in join_entry.cached.values() {
-                                // check join condition
-                                let ok = if let Some(ref mut cond) = self.condition {
-                                    let concat_row = left_row.chain(&right_row).into_owned_row();
-                                    cond.eval_row_infallible(&concat_row)
-                                        .await
-                                        .map(|s| *s.as_bool())
-                                        .unwrap_or(false)
-                                } else {
-                                    true
+                    // Joined result without evaluating non-lookup conditions.
+                    let st1 = {
+                        #[try_stream]
+                        async {
+                            #[allow(unreachable_code)]
+                            #[allow(clippy::diverging_sub_expression)]
+                            if false {
+                                return unreachable!("type hints only") as StreamExecutorResult<_>;
+                            }
```

Comment:
Is it possible to extract a method for this async block? The hack here looks ugly🤣

Reply:
Prefer to keep the hack code in a small scope and use it only needed.

Reply:
I think this is very acceptable. I like the simple and stupid way